shared_model/backend/protobuf/proposal.hpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_PROTO_PROPOSAL_HPP
+#define IROHA_SHARED_MODEL_PROTO_PROPOSAL_HPP
+
+#include "interfaces/proposal.hpp"
+#include "interfaces/transaction.hpp"
+
+#include <boost/range/numeric.hpp>
+#include "common_objects/trivial_proto.hpp"
+#include "model/proposal.hpp"
+
+#include "block.pb.h"
+#include "ordering.pb.h"
+#include "utils/lazy_initializer.hpp"
+
+#include "transaction.hpp"
+namespace shared_model {
+  namespace proto {
+    class Proposal FINAL
```

Comment:
This class is not template, what is the reason of putting definitions in hpp file?

hpp file must contain only declarations